src/configure/clients/github/githubClient.ts

```diff
@@ -1,69 +1,150 @@
+                url: Url,
+                headers: {
+                    "Content-Type": "application/json",
+                    "Authorization": "Bearer " + this.patToken,
+                    "User-Agent": UserAgent
+                },
+                method: 'POST',
+                body: {
+                    "name": repoName,
+                    "description": "Repo created from VScode",
+                    "homepage": "https://github.com",
+                    "private": false,
+                    "has_issues": true,
+                    "has_projects": true,
+                    "has_wiki": true
+                },
+                deserializationMapper: null,
+                serializationMapper: null
+            })
+                .then((detail: GitHubRepo) => {
+                    return detail;
+                }).catch(error=>{
+                        return null;
+                });
+        }
+        catch(error){
+            return null;
+        }
     }
 
-    private async _getGitHubSecretKey() : Promise<GitHubSecretKey> {
-        let request = <UrlBasedRequestPrepareOptions> {
-                url: GitHubProvider.getFormattedGitHubApiUrlBase(this.url) + "/actions/secrets/public-key",
+    public async listOrganizations(PATtoken: string, forceRefresh?: boolean): Promise<void | GitHubOrganization[]> {
+        if (!this.listOrgPromise || forceRefresh) {
+            let restClient = new RestClient();
+            this.listOrgPromise = restClient.sendRequest(<UrlBasedRequestPrepareOptions>{
+                url: "https://api.github.com/user/orgs",
                 method: 'GET',
                 headers: {
+                    "Authorization": "Bearer " + PATtoken,
                     "User-Agent": UserAgent,
                     "Content-Type": "application/json",
-                    "Authorization": "Bearer " + this.patToken,
                     "Accept": "*/*"
                 },
-                serializationMapper: null,
-                deserializationMapper: null
-            }
-        let restClient = new RestClient();
-        return (await restClient.sendRequest(request)) as GitHubSecretKey;
+                deserializationMapper: null,
+                serializationMapper: null
+            })
+                .then((organizations: Array<GitHubOrganization>) => {
+                    organizations = organizations.sort((org1, org2) => stringCompareFunction(org1.login, org2.login));
+                    return organizations;
+                });
+        }
+        return this.listOrgPromise; 
     }
 
-    private async _setGithubSecret(secretName: string, key_id: string, encrypted_secret: string): Promise<void> {
+    public async generateGitHubRepository(orgName: string, localPath: string): Promise<GitHubRepo | void>{
+        let repoName = localPath.substring(localPath.lastIndexOf("\\")+1).replace("*","-"); //need to include all the special characters here
```

Comment:
why do you want to include special chars? why not remove all non-alphanumeric characters?

Reply:
I have removed the 'replace' function now since GitHub automatically handles the special cases.